src/components/TxDialog/TxDialog.component.tsx

```diff
@@ -144,3 +161,93 @@ export const SubmitStatusDialog = ({
     />
   );
 };
+
+const finishStep: StepData<'finish'> = {
+  name: 'finish',
+  label: 'finish',
+};
+
+const CallStepper = <Operations extends string>({
+  steps,
+  status,
+  currentStep,
+}: CallStepperProps<Operations>) => {
+  type AllSteps = Operations | 'finish';
```

Comment:
we can move to a types file :D

Reply:
We would need to convert the type to be generic in that case. If it's declared here we have a direct access to the `Operations` parameter